pkg/cmd/certregenerationcontroller/certregenerationcontroller.go

```diff
@@ -0,0 +1,295 @@
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/certrotationcontroller"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+)
+
+type Options struct {
+	KubeconfigPath string
+	TLSServerName  string
+
+	LeaderelectionLeaseDuration time.Duration
+	LeaderelectionRenewDeadline time.Duration
+	LeaderelectionRetryPeriod   time.Duration
+
+	restConfig   *restclient.Config
+	kubeClient   kubernetes.Interface
+	configClient configeversionedclient.Interface
+}
+
+func NewCertRegenerationControllerCommand() *cobra.Command {
+	o := &Options{
+		KubeconfigPath: "",
+		TLSServerName:  "",
+
+		LeaderelectionLeaseDuration: 60 * time.Second,
+		LeaderelectionRenewDeadline: 35 * time.Second,
+		LeaderelectionRetryPeriod:   10 * time.Second,
+	}
+
+	cmd := &cobra.Command{
+		Use: "cert-regeneration-controller",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	cmd.PersistentFlags().StringVarP(&o.KubeconfigPath, "kubeconfig", "", o.KubeconfigPath, "Path to the kubeconfig file")
+	cmd.PersistentFlags().StringVarP(&o.TLSServerName, "tls-server-name", "", o.TLSServerName, "The SNI hostname to set for the server in kubeconfig")
+
+	cmd.PersistentFlags().DurationVar(&o.LeaderelectionLeaseDuration, "leaderelection-lease-duration", o.LeaderelectionLeaseDuration, "LeaseDuration is the duration that non-leader candidates will wait to force acquire leadership.")
+	cmd.PersistentFlags().DurationVar(&o.LeaderelectionRenewDeadline, "leaderelection-renew-deadline", o.LeaderelectionRenewDeadline, "RenewDeadline is the duration that the acting master will retry refreshing leadership before giving up.")
+	cmd.PersistentFlags().DurationVar(&o.LeaderelectionRetryPeriod, "leaderelection-retry-period", o.LeaderelectionRetryPeriod, "RetryPeriod is the duration the LeaderElector clients should wait between tries of actions.")
+
+	return cmd
+}
+
+func (o *Options) Validate() error {
+	return nil
+}
+
+func (o *Options) Complete() error {
+	var err error
+
+	if len(o.KubeconfigPath) != 0 {
+		klog.V(1).Infof("Using kubeconfig %q.", o.KubeconfigPath)
+		o.restConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: o.KubeconfigPath}, &clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			return fmt.Errorf("can't create config from kubeConfigPath %q: %w", o.KubeconfigPath, err)
+		}
+	} else {
+		klog.V(1).Infof("No kubeconfig specified, using InClusterConfig.")
+		o.restConfig, err = restclient.InClusterConfig()
+		if err != nil {
+			return fmt.Errorf("can't create InClusterConfig: %w", err)
+		}
+	}
+
+	if len(o.TLSServerName) != 0 {
```

Comment:
you need to explain this.  not everyone is familiar with SNI and why this matters.

Reply:
done